pkg/rpc/methods.go

```diff
@@ -81,15 +81,18 @@ func (c *Client) ensureGenesisMatched(ctx context.Context) error {
 
 // WaitTillL2ExecutionEngineSynced keeps waiting until the L2 execution engine is fully synced.
 func (c *Client) WaitTillL2ExecutionEngineSynced(ctx context.Context) error {
-	if ctx.Err() != nil {
+	if ctx.Err() != nil && !errors.Is(ctx.Err(), context.DeadlineExceeded) {
 		return ctx.Err()
 	}
+	newCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	start := time.Now()
 	return backoff.Retry(
 		func() error {
-			if ctx.Err() != nil {
+			if ctx.Err() != nil && !errors.Is(ctx.Err(), context.DeadlineExceeded) {
```

Comment:
```suggestion
			if ctx.Err() != nil && !errors.Is(ctx.Err(), context.Background()) {
```

Seems this should also be ok?